### PR TITLE
Issue #18572 Add ReadonlySpan CopyTo test

### LINKED ISSUE #18572: Add ReadOnlySpan CopyTo tests that already exist for Span
cc @WinCPP, @shiftylogic 

An example of this would be the test recently added here: https://github.com/dotnet/corefx/pull/18286

We want these and possibly other such tests added to RoS from:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/Span/CopyTo.cs#L205

To here:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/ReadOnlySpan/CopyTo.cs#L123

### PR DESCRIPTION
**Pull request by WinCPP**
Fixes #18572.

@ahsonkhan @shiftylogic Please review.

### CODE REVIEW

File: src/System.Memory/tests/ReadOnlySpan/CopyTo.cs [hunk 1]
```diff
@@ -121,5 +122,68 @@ public static void Overlapping2()
...
+            {
+                try
+                {
+                    allocatedFirst = AllocationHelper.TryAllocNative((IntPtr)bufferSize, out memBlockFirst);
+                    allocatedSecond = AllocationHelper.TryAllocNative((IntPtr)bufferSize, out memBlockSecond);
+
+                    if (allocatedFirst && allocatedSecond)
+                    {
+                        ref Guid memoryFirst = ref Unsafe.AsRef<Guid>(memBlockFirst.ToPointer());
+                        var spanFirst = new Span<Guid>(memBlockFirst.ToPointer(), GuidCount);
```
Comment (ahsonkhan) [inline src/System.Memory/tests/ReadOnlySpan/CopyTo.cs]: ReadOnlySpan?

Comment (WinCPP) [inline src/System.Memory/tests/ReadOnlySpan/CopyTo.cs]: oops... sorry... effect of sleeping at work :(

**Comment by ahsonkhan** [comment]
LGTM

File: src/System.Memory/tests/ReadOnlySpan/CopyTo.cs [hunk 2]
```diff
@@ -121,5 +122,68 @@ public static void Overlapping2()
...
+        // the CopyTo method performs copy in chunks of size 4GB (uint.MaxValue) with final iteration copying
+        // the residual chunk of size (bufferSize % 4GB). The inputs sizes to this method, 4GB and 4GB+256B,
+        // test the two size selection paths in CoptyTo method - memory size that is multiple of 4GB or,
+        // a multiple of 4GB + some more size.
+        [Theory]
+        [OuterLoop]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [InlineData(4L * 1024L * 1024L * 1024L)]
+        [InlineData((4L * 1024L * 1024L * 1024L) + 256)]
+        public static void CopyToLargeSizeTest(long bufferSize)
```
Comment (safern) [inline src/System.Memory/tests/ReadOnlySpan/CopyTo.cs]: Does this test pass on Desktop?

Could you please verify that by doing:
`build.cmd -framework:netfx`
`msbuild <path-to-test-csproj> /t:rebuildandtest /p:targetgroup=netfx`

Comment (WinCPP) [inline src/System.Memory/tests/ReadOnlySpan/CopyTo.cs]: @safern Yes the test passes on desktop using above two commands...

Curious: How is this different from running the same test without `netfx` which also passed? Was that not dekstop then?

Comment (ahsonkhan) [inline src/System.Memory/tests/ReadOnlySpan/CopyTo.cs]: The default is netcoreapp2.0 which is using Fast Span in mscorlib.

**Pull request merged**